synapse_bind_sydent/__init__.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2022 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from typing import Any, Dict
+from urllib.parse import urlparse
+
+import attr
+from synapse.module_api import ModuleApi
+from synapse.module_api.errors import ConfigError
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True, frozen=True)
+class SydentBinderConfig:
+    sydent_base_url: str
+
+
+class SydentBinder:
+    def __init__(self, config: SydentBinderConfig, api: ModuleApi) -> None:
+        # Keep a reference to the config and Module API
+        self._api = api
+        self._config = config
+
+        self._http_client = api.http_client
+
+        self._sydent_bind_url = (
+            f"{config.sydent_base_url}/_matrix/identity/internal/bind"
+        )
+
+        scheme = urlparse(config.sydent_base_url).scheme
+        self._sydent_host = config.sydent_base_url.replace(f"{scheme}://", "")
+
+        self._api.register_third_party_rules_callbacks(
+            on_threepid_bind=self.on_threepid_bind,
+        )
+
+    @staticmethod
+    def parse_config(config: Dict[str, Any]) -> SydentBinderConfig:
+        if "sydent_base_url" not in config or not isinstance(
+            config["sydent_base_url"], str
+        ):
+            raise ConfigError("sydent_base_url needs to be a string")
+
+        if not config["sydent_base_url"].startswith("http://") and not config[
+            "sydent_base_url"
+        ].startswith("https://"):
```

Comment:
```suggestion
        if urlparse(config["sydent_base_url"]).scheme not in ("http", "https"):
```

(This also lets you type `HTTPS://` if you really want to, not that it really matters)